### PR TITLE
fix: add with_universe_domain

### PR DESCRIPTION
**Pull request by arithmetic1728**

### CODE REVIEW

**Review by BigTailWolf — APPROVED** [review]

**Review by TimurSadykov — COMMENTED** [review]
No issue with the code.. but let's discuss this on next weekly sync. Maybe we can drop create_with_universe_domain helpers.

File: google/auth/external_account.py [hunk 1]
```diff
@@ -415,6 +415,22 @@ def with_token_uri(self, token_uri):
         new_cred._metrics_options = self._metrics_options
         return new_cred
 
+    def with_universe_domain(self, universe_domain):
```
[the remark this answers is not in the document]
Comment (arithmetic1728) [reply google/auth/external_account.py:418]: `with_*` is the regular pattern for google-auth lib, for example, `with_quota_project_id`, `with_scopes`, `with_token_uri` etc. Builder/Constructor pattern is specific for Java auth lib, python lib doesn't do this.

File: google/oauth2/credentials.py [hunk 2]
```diff
@@ -126,6 +128,9 @@ def __init__(
             granted_scopes (Optional[Sequence[str]]): The scopes that were consented/granted by the user.
                 This could be different from the requested scopes and it could be empty if granted
                 and requested scopes were same.
+            trust_boundary (str): String representation of trust boundary meta.
```
Comment (clundin25) [inline google/oauth2/credentials.py:131]: Is this a merge error?

**Review by clundin25 — APPROVED** [review]

File: google/oauth2/credentials.py (hunk 2 above)
Comment (arithmetic1728) [reply google/oauth2/credentials.py:131]: no, when I added the universe_domain doc string I noticed that the trust_boundary doc string was missing, so I just added it as well.

File: google/auth/external_account.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimurSadykov) [reply google/auth/external_account.py:418]: Ack, LGTM (passing by)

**Pull request merged**